src/js/new-visualization-modal.js

```diff
@@ -2,41 +2,34 @@ import * as app from './app';
 
 import dataQuery from '../data/data_query.sql';
 
+let queryResults = null;
 // handle modal submit
 function onNewVisSubmit() {
   // remove old warnings
-  $('#invalid-query-results-warning').hide();
-  $('#empty-query-results-warning').hide();
-
-  // get query results
-  let queryResults = $('textarea#data-input').val().trim();
-
-  // validate results
-  if (queryResults === '') {
+  hideAllWarnings();
+  if (!queryResults) {
     $('#empty-query-results-warning').show();
-    return;
   }
 
-  // parse the text into a hierarchy array
-  if (queryResults.startsWith('GRANTS')) {
-    const index = queryResults.indexOf('[');
-    queryResults = queryResults.substring(index);
-  } else if (!queryResults.startsWith('[')) {
-    $('#invalid-query-results-warning').show();
-    return;
-  }
-
-  // plot the visualization
-  $('#create-vis-modal').modal('toggle');
   try {
-    app.init(JSON.parse(queryResults));
+    app.init(queryResults);
+    $('#create-vis-modal').modal('toggle');
   } catch (error) {
+    $('#invalid-query-results-warning').show();
     alert(
```

Comment:
remove alerts. Handle this without alerts

Reply:
Done